client/my-sites/themes/recommended-themes.jsx

```diff
@@ -1,52 +1,42 @@
-import { Component } from 'react';
-import { connect } from 'react-redux';
+import { useLayoutEffect } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import { useGutenbergFSESettingsQuery } from 'calypso/data/gutenberg/fse-settings-query';
 import { getRecommendedThemes } from 'calypso/state/themes/actions';
 import {
 	getRecommendedThemes as getRecommendedThemesSelector,
 	areRecommendedThemesLoading,
-	getRecommendedThemesFilter,
 } from 'calypso/state/themes/selectors';
-import { getSelectedSiteId } from 'calypso/state/ui/selectors';
 import { ConnectedThemesSelection } from './themes-selection';
 
-class RecommendedThemes extends Component {
-	componentDidMount() {
-		if ( ! this.props.customizedThemesList.length ) {
-			this.fetchThemes();
-		}
-	}
+const RecommendedThemes = ( props ) => {
+	const dispatch = useDispatch();
+	const { siteId } = props;
+	const { isLoading: isLoadingGutenbergFSESettings, data } = useGutenbergFSESettingsQuery( siteId );
 
-	componentDidUpdate( prevProps ) {
-		// Wait until rec themes to be loaded to scroll to search input if its in use.
-		const { isLoading, scrollToSearchInput, filter } = this.props;
-		if ( prevProps.isLoading !== isLoading && isLoading === false ) {
-			scrollToSearchInput();
```

Comment:
This is not necessary since any keystroke on the filter's input would move the user to all themes.